File: Introduction_to_numerical_programming_using_Python_and_CPP_Beu/Ch11/C/P11-MC-2D.cpp
// Monte Carlo calculation of the unit circle area
#include <stdio.h>
#include <math.h>
#include "random.h"

int main()
{
   double fi, s, sigma, x, y;
   long i, n, ni;

   printf("n = "); scanf("%li",&n);              // number of sampling points

   seed();

   ni = 0;                                       // number of interior points
   for (i=1; i<=n; i++) {
      x = random(); y = random();
      if (x*x + y*y <= 1e0) ni++;                       // add interior point
   }
   fi = (double) ni/n;                         // fraction of interior points
   s = 4e0 * fi;                                                  // integral
   sigma = 4e0*sqrt((fi - fi*fi)/n);                    // standard deviation
   printf("Unit circle area = %f +/- %f\n",s,sigma);
}

File: Introduction_to_numerical_programming_using_Python_and_CPP_Beu/Ch11/Python/P11-MC-2D.py
# Monte Carlo calculation of the unit circle area
from math import *
from random import *

# main

n = eval(input("n = "))                           # number of sampling points

seed()

ni = 0                                            # number of interior points
for i in range(1,n+1):
   x = random(); y = random()
   if (x*x + y*y <= 1e0): ni += 1                        # add interior point

fi = ni/n                                       # fraction of interior points
s = 4e0 * fi                                                       # integral
sigma = 4e0 * sqrt((fi - fi*fi)/n)                       # standard deviation
print("Unit circle area = ",s," +/- ",sigma)
